File: leetcodePython/tree/1161.maximum-level-sum-of-a-binary-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def getChildren(self, root, result):
        if root.left:
            result.append(root.left)
        if root.right:
            result.append(root.right)

    def maxLevelSum(self, root: TreeNode) -> int:
        if not root:
            return 0

        stack = [root]
        solution = [root.val]
        while stack:
            temp = []
            while stack:
                self.getChildren(stack.pop(0), temp)
            cnt = 0
            for i in temp:
                cnt += i.val
            solution.append(cnt)
            stack += temp

        return solution.index(max(solution))+1

File: leetcodeJS/tree/1161.maximum-level-sum-of-a-binary-tree.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number}
 */

const getChildren = (root) => {
  if (root.left && root.right) return [root.left, root.right];
  if (root.left && !root.right) return [root.left];
  return [root.right];
};

var maxLevelSum = function (root) {
  if (!root) return;
  let solution = [root.val];
  let stack = [root];
  while (stack.length) {
    let temp = [];
    while (stack.length) {
      let cur = stack.shift();
      solution[solution.length - 1] += cur.val;
      temp = [...temp, ...getChildren(cur)];
    }
    stack = [];
    for (i of temp) {
      if (i) stack.push(i);
    }
    if (stack.length) solution.push(0);
  }
  return solution.indexOf(Math.max(...solution)) + 1;
};
